robot/client/client_test.go

```diff
@@ -859,21 +883,24 @@ func TestClientUnaryDisconnectHandler(t *testing.T) {
 			info *grpc.UnaryServerInfo,
 			handler grpc.UnaryHandler,
 		) (interface{}, error) {
+			// Allow a single GetMachineStatus through; return `io.ErrClosedPipe`
+			// after that.
```

Comment:
Left a comment here to better describe what this test is doing. We also have to call `handler(ctx, req)` below, now, as we do want to invoke the injected `MachineStatus` to understand that the robot is running.